### PR TITLE
Various fixes/updates

### LINKED ISSUE #4: New path for Marshmallow
For Marshmallow the paths of the odex files is changed to "oat"/arch/X.odex

### PR DESCRIPTION
**Pull request by mfonville**
Should fix #4 
Introduces multi-arch compatibility

### CODE REVIEW

File: odex2apk.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 OAT2DEX = os.getenv("OAT2DEX", os.path.join(_dirname, "oat2dex.jar"))
 
 # Supported architectures (first match will be used)
-architectures = ["arm", "arm64", "x86", "x86_64"]
+architectures = ["x86_64", "x86", "arm64", "arm"]
```
Comment (Lekensteyn) [inline odex2apk.py:41]: This changes the detection order, so if both `arm64` and `arm` exists, then the former will now be used. Is this really the right logic? Would it be more accurate if the `detect_arch` function is modified to check for presence of another file?

Comment (mfonville) [inline odex2apk.py:41]: For all Nexus images this should work correctly. The Nexus image's APKs are all compiled for 1 architecture only. The 32-bit folder presence on 64-bit images is only to enable compatibility for running third party 32-bit apps on the 64-bit device.

Comment (Lekensteyn) [inline odex2apk.py:41]: Ah ok, so it is done to ensure that 64-bit folders are detected before their 32-bit counterparts? So `arm64, arm, x86_64, x86` would also be sufficient?

**Comment by mfonville** [comment]
Yes, for the detection order.
But best is the order in suggested, because that is how Android and the Play store also handle it internally (for e.g. vercodes). Because platforms can also mix x86 and arm with libhoudini.

**Comment by Lekensteyn** [comment]
Merged, thanks for your work and explanations!

**Pull request merged**